### PR TITLE
Remove remaining arbitrary restrictions

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit removes all (most) of the remaining code limiting xenomorphs
that have has_fine_manipulation on an entirely arbitrary basis. Xenomorphs
with has_fine_manipulation may now interact with NanoUI, and will only
break computers on non-help intent. The can also tear down walls, since
walls used an ishuman check in place of an isAdvancedToolUser check as
they should have.

Minor refactor to machinery.dm as well; Instead of the slightly
(completely) insane attack_hand restriction to humans and silicons (it was
a multi-line IF, why), it now checks user.IsAdvancedToolUser(); This means
restrictions on monkies will actually function properly now.

Also changed:
 - Paper now works on a say_understands basis.
  - This means that anything with universal speak/understand, or anything with Galactic Common can read papers. Technically allows simple_animals to read paper when they could not before, but whatever.

@Markolie You know nanoUI pretty well; Going to request a review to make sure I didn't fuck anything up too badly.

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -341,10 +341,11 @@ Class Procs:
 		return 1
 	if(user.lying || user.stat)
 		return 1
-	if ( ! (istype(usr, /mob/living/carbon/human) || \
-			istype(usr, /mob/living/silicon)))
-		usr << "\red You don't have the dexterity to do this!"
+
+	if(!user.IsAdvancedToolUser())
+		user << "\red You don't have the dexterity to do this!"
```
Comment (Markolie) [inline code/game/machinery/machinery.dm]: <span class='warning'>

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -384,7 +384,7 @@ REAGENT SCANNER
 	if (crit_fail)
 		user << "\red This device has critically failed and is no longer functional!"
 		return
-	if (!(istype(user, /mob/living/carbon/human) || ticker) && ticker.mode.name != "monkey")
+	if (!user.IsAdvancedToolUser())
 		user << "\red You don't have the dexterity to do this!"
```
Comment (Markolie) [inline code/game/objects/items/devices/scanners.dm]: ```<span class='warning'>```

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -441,7 +441,7 @@ REAGENT SCANNER
 /obj/item/device/reagent_scanner/afterattack(obj/O, mob/user as mob)
 	if (user.stat)
 		return
-	if (!(istype(user, /mob/living/carbon/human) || ticker) && ticker.mode.name != "monkey")
+	if (!user.IsAdvancedToolUser())
 		user << "\red You don't have the dexterity to do this!"
```
Comment (Markolie) [inline code/game/objects/items/devices/scanners.dm]: ```<span class='warning'>```

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -169,8 +169,8 @@
 	afterattack(atom/target as mob|obj|turf|area, mob/user as mob, flag)
 		if (flag)
 			return
-		if (!(istype(usr, /mob/living/carbon/human) || ticker) && ticker.mode.name != "monkey")
-			usr << "\red You don't have the dexterity to do this!"
+		if (!user.IsAdvancedToolUser())
+			user << "\red You don't have the dexterity to do this!"
```
Comment (Markolie) [inline code/game/objects/items/toys.dm]: ```<span class='warning'>```

File: code/modules/nano/interaction/default.dm [hunk 5]
```diff
@@ -102,3 +102,10 @@
 		. = min(., shared_living_nano_distance(src_object))
 		if(. == STATUS_UPDATE && (TK in mutations))	// If we have telekinesis and remain close enough, allow interaction.
 			return STATUS_INTERACTIVE
+
+/mob/living/carbon/alien/default_can_use_topic(var/src_object)
+	. = shared_nano_interaction(src_object)
+	if(. != STATUS_CLOSE)
+		. = min(., shared_living_nano_distance(src_object))
+		if(!IsAdvancedToolUser())
+			. = STATUS_UPDATE
```
Comment (Markolie) [inline code/modules/nano/interaction/default.dm]: I'm not sure whether we should go with STATUS_UPDATE or STATUS_CLOSE here, though I'm leaning towards STATUS_CLOSE. It depends on whether or not we want Xenos to be able to still read monitors and other things when they aren't an advanced tool user.

**Comment by Markolie** [comment]
Consider adding a IsAdvancedToolUser check for simple animals in NanoUI as well, so that admin animals can also access the interfaces.

File: code/modules/nano/interaction/default.dm (hunk 5 above)
Comment (tigercat2000) [inline code/modules/nano/interaction/default.dm]: They can't even look at computer screens if they don't have advanced tool user status; They just smash the console.

**Comment by tigercat2000** [comment]
@Markolie Simple animals actually don't have any way to pass AdvancedToolUser checks.

**Comment by Ultimate-Chimera** [comment]
Ugh, why with all the xenos stuff? Tiger this is making me go insane.

**Comment by Ultimate-Chimera** [comment]
You just put in all these new things for xenos so just YOU get to use them, it's unfair.

**Comment by tigercat2000** [comment]
@Ultimate-Chimera Things need to be fixed, Chimera. I rather wish for the xenomorph gamemode back, which definately isn't going to happen without a lot of work to xenomorphs.
You really need to work on not obsessing over things.

**Comment by Spacemanspark** [comment]
To be honest, I think adding /tg/ xenos was a step in the wrong direction, no offense to anyone.

**Comment by Ultimate-Chimera** [comment]
Adding bay xenos would have been worse.

**Comment by tigercat2000** [comment]
@Spacemanspark Well, given that the choice was -tg- xenos (actually xenos) or bay xenos (literally fucking ERPfest, if zuhayr gets his way)

**Comment by Spacemanspark** [comment]
What about CM xenos? :^)

**Comment by tigercat2000** [comment]
CM isn't open source last I checked

**Comment by Spacemanspark** [comment]
Well, the old version is.
But I was mostly joking.

**Comment by Ultimate-Chimera** [comment]
@tigercat2000 I'd love to but it is hard when you are a xeno all the time and nobody else gets to be a friendly xeno seems unfair and it bugs me, because when I'm a xeno I get that good feeling mentally that I can not explain, but basically when I ignore it, I start becoming very pissy and rude, etc. I have tried to get over my obsession by ignoring it and it ended up badly, what I did with my last obsession was over-do it till I got bored of it, which might work. But just delaying that is making it worse, I am not even joking. Besides, all these new features and I do not even get to test/see them, only you.

**Comment by Ultimate-Chimera** [comment]
My last one lasted only a year and I was back to normal, this one lasted OVER a year, so that's a problem. And I was fine for a bit, but then I started to notice some person taking this xenos suit from a certain place, which somehow triggered this obsession, I learnt where it was and started to go for it, and then I donated for it. After a bit I changed my name on steam, profile picture, desktop background, etc.

**Comment by Ultimate-Chimera** [comment]
So the cause of it was basically seeing somebody else as a xeno, so this is kind of making it worse, you  understand now?

**Comment by Robertha89** [comment]
@Ultimate-Chimera if you are the only xeno, ask an admin, over half of them are reasonable enough, and knows friendly xeno better then hostile xeno. Alternativly, kill the other xenos, then become friendly, or even talk to an admin about being spawned in as a friendly xeno, i mean technically, you can make xenos wear hats, and thus you can have a xeno hunter as HoS or even HoP, cause now xenos will be able to pick up stuff again.

**Comment by Ultimate-Chimera** [comment]
This is exactly why I get really pissed when I see other people as a xeno.

**Comment by TheDZD** [comment]
Keep the PR comments discussion relevant to the actual content of the PR, and not accusations made against the coder.

**Comment by Ultimate-Chimera** [comment]
Dude, I HAVE asked, and they do not wanna do it because think of the OTHER people, it is an EVENT and admins can't just straight up do it, they need to know if others will enjoy it, etc. Also sorry DZD, I will. I support the animals fix with paper, etc. :+1:

**Comment by Dumbdumn5** [comment]
@TheDZD Has a point, keep it civil here.

**Comment by Ultimate-Chimera** [comment]
I have a question, will all simple mobs be able to understand paper now? And will they be able to read other things with text like for example, a fire alarm? Or will it all be jumbled up mess still?

**Comment by tigercat2000** [comment]
Change only affects paper.

**Comment by Ultimate-Chimera** [comment]
Would it be possible to make it affect other things as well?

**Comment by tigercat2000** [comment]
I'm fairly sure fire alarms and the like use NanoUI, and default behaviour for things not implemented specifically (like simple animals) is just to close the window immediately

**Comment by Ultimate-Chimera** [comment]
Oh, so that's why some windows close instantly when you open them? Like I was a carded AI and for example, the airlock control UI started closing and opening and it was so annoying as it started to make the window dissapear at the bottom of the screen if not moved.

**Comment by Ultimate-Chimera** [comment]
Oh, honestly I think having xenos have fine_manipulation by default is kind of OP, as they can pick up guns and shoot everyone. :P

**Comment by tigercat2000** [comment]
They don't have it by default, though?

**Comment by Ultimate-Chimera** [comment]
But this PR basically removes their limiting, right?

**Comment by Ultimate-Chimera** [comment]
Oh wait, I read it wrong... If they DO have has_fine_manipulation, it does not remove the var.

**Comment by Ultimate-Chimera** [comment]
:+1:

File: code/modules/nano/interaction/default.dm (hunk 5 above)
Comment (Markolie) [inline code/modules/nano/interaction/default.dm]: It's best to make STATUS_UPDATE --> STATUS_CLOSE then.

**Comment by tigercat2000** [comment]
This PR removes a bunch of hardcoded type checks that should be checking if they have fine manipulation; since xenomorphs with fine manipulation are always adminboozed, they shouldn't be stuck to not using computers or bibles etc

**Comment by Ultimate-Chimera** [comment]
Makes sense.

**Comment by Zuhayr** [comment]
>(literally fucking ERPfest, if zuhayr gets his way)

@tigercat2000 ?

**Comment by tigercat2000** [comment]
@Zuhayr Baystation12/Baystation12#10995 
2mucherp5me

**Comment by Zuhayr** [comment]
They're literal penis monsters. That PR removes literal alien face rape. If you're going to talk shit at least use accurate criticism.

**Comment by tigercat2000** [comment]
@Zuhayr relax I don't care about your alien changes, but they definitely go against the vibe of Alien

**Comment by Zuhayr** [comment]
If you don't care about my changes don't bring them up in an unrelated PR on a separate repo and then use them to imply I'm wilfully forcing fucking ERP into the game.

I have absolutely no investment in the changes and don't care what you think about them, but having my name dropped solely to lie about what I'm doing is not appreciated.

**Comment by Belario** [comment]
... So Aliens can finally mess with APCs? :+1:

@Zuhayr 
zjhagtalhh6qcjymr4ps

**Comment by Spacemanspark** [comment]
Oh dear.

**Comment by tigercat2000** [comment]
yay

**Pull request merged**